forward/testing/unit_tests.lua

```diff
@@ -34,26 +37,50 @@ end
 -- output_generators (array of functions) -- Generators that produce
 --    the expected output from Spike. These are functions that take
 --    in the backend IP address and produce a packet.
-function UnitTests:run_test(test_name, input_packets, output_generators, valid_backend_addrs)
-   valid_backend_addrs =
-      valid_backend_addrs or self.network_config.backend_addrs
-
+function UnitTests:run_test(test_name, input_packets, output_generators, backends)
    print("Running test: "..test_name)
    self.stream_app:init(input_packets)
 
+   self.backend_tracker:set_backends(backends)
    self.expected_output_generator_app:init(
-      output_generators, valid_backend_addrs)
+      output_generators, backends)
    local expected_num_output_packets = #output_generators
 
+   local err = nil
    local test_start_time = os.clock()
+   local flush_counter = -1
    engine.main({done = function()
-      assert(os.clock() - test_start_time < 0.1,
-         "Test timed out. Possibly too many packets were dropped.")
-      return #self.out_collect_app.packets ==
+      if os.clock() - test_start_time > 0.1 then
+         err = "Test timed out. Possibly too many packets were dropped."
```

Comment:
What causes this to happen?

Reply:
We can't differentiate between rewriting dropping packets when it's not supposed to and it taking a long time (unless we break the rewriting black box, or explicitly count engine breaths). Either of these would trigger that error. If we have longer tests in the future, we can always make the timeout configurable.